src/raylib/textures/Texture2D.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+** EPITECH PROJECT, 2022
+** Bomberman-Global-Offensive
+** File description:
+** Textures
+*/
+
+#ifndef TEXTURES_HPP_
+#define TEXTURES_HPP_
+
+#include "raylib/core/Vector2.h"
+
+extern "C"
+{
+#include "raylib.h"
+}
+
+namespace raylib
+{
+    namespace textures
+    {
+        class Texture2D {
+          public:
+            /// This function takes in a position and a file name and sets the position
+            /// and file name of the texture to the values passed in
+            ///
+            ///@param position The position of the texture on the screen.
+            ///@param fileName The name of the file that contains the texture.
+            Texture2D(
+                Vector2 position =
+                {
+                    0;
+                    0
+                },
+                char *fileName);
+
+            /// The destructor for the Texture2D class
+            ~Texture2D();
+
+            /// Sets the position of the texture
+            ///
+            ///@param position The position of the texture.
+            setPosition(Vector2 position);
+
+            /// The function `loadTexture()` loads a texture from a file
+            Texture2D loadTexture();
+
+            /// Loads a texture from an Image object
+            ///
+            ///@param image The image to load the texture from.
+            void loadTextureFromImage(Image image);
+
+            /// UnloadTexture() is a function that unloads a texture from GPU memory
+            void unloadTexture();
+
+            /// It updates the texture with the given pixels.
+            ///
+            ///@param pixels The pixel data to update the texture with.
+            void updateTexture(const void *pixels);
```

Comment:
`std::span<T>` would be better suited here, the type of `T` may be `char`, or `uint32_t`